src/reducers/menuReducer.js

```diff
@@ -0,0 +1,41 @@
+const initialState = {
+  menu: []
+};
+
+export default function (state = initialState, action) {
+  // if (action.type === "MENU_LOAD") {
+  //   return {
+  //     ...state,
+  //     menu: action.payload
+  //   };
+  // }
+
+  switch (action.type) {
+    case 'MENU_LOAD':
+      return {
+        ...state,
+        menu: action.payload
+      };
+    case 'ADD_MENU':
+      return {
+        ...state,
+        menu: [...state.menu, action.payload]
+      };
+    case 'DELETE_MENU':
+      const updatedMenu = [...state.menu]   // copy
+      const deletedArray = updatedMenu.filter((updatedMenus) => {
+        if (updatedMenus.id === action.payload) {
+          return false; //delete
```

Comment:
If the condition is true it will give the array with deleted id or else it will give the entire array